### PR TITLE
Add a GitHub Action to test the benchmarks

### PR DESCRIPTION
**Pull request by inaki-amatria**
This PR adds a GitHub Action workflow to compile and run the benchmarks across multiple operating systems (Ubuntu 22.04, macOS 14, and Windows 2022). The workflow ensures that the codebase compiles and runs successfully on these platforms.

### CODE REVIEW

**Review by jgonzac — APPROVED** [review]

**Review by alvrogd — COMMENTED** [review]
Can we check in advance how the Action will behave? In particular, I'm wondering if the different `os` configurations provide all necessary tools (e.g., compilers) to run both C and Fortran benchmarks. If not, how do we configure them?

**Comment by inaki-amatria** [comment]
> Can we check in advance how the Action will behave? In particular, I'm wondering if the different `os` configurations provide all necessary tools (e.g., compilers) to run both C and Fortran benchmarks, thus ensuring full coverage. If not, how do we configure them?

See https://github.com/codee-com/open-catalog/actions/runs/9316409557, please!

File: .github/workflows/Compile-and-run.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        os: [ ubuntu-22.04, macos-14, windows-2022 ]
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+      - name: Run build and benchmark script
+        run: python Benchmark/run-benchmarks.py -- -G"Unix Makefiles"
```
Comment (daniel-otero) [inline .github/workflows/Compile-and-run.yml:24]: I know that GH Runner are not a good platform to tests this, but it is so nice to see the benchmarks there... How hard would it be to split this step in "build" and "benchmark"?

**Comment by alvrogd** [comment]
> > Can we check in advance how the Action will behave? In particular, I'm wondering if the different `os` configurations provide all necessary tools (e.g., compilers) to run both C and Fortran benchmarks, thus ensuring full coverage. If not, how do we configure them?
> 
> See https://github.com/codee-com/open-catalog/actions/runs/9316409557, please!

Sorry for the oversight :sweat: 

Looks great :)

**Review by alvrogd — APPROVED** [review]

File: .github/workflows/Compile-and-run.yml (hunk 1 above)
Comment (daniel-otero) [inline .github/workflows/Compile-and-run.yml:24]: Isn't ninja installed? :rofl: 

```suggestion
        run: python Benchmark/run-benchmarks.py -- -G"Ninja"
```

**Review by daniel-otero — APPROVED** [review]
Looks great! Didn't expect all C++ utilities to be implicitly present.

File: .github/workflows/Compile-and-run.yml (hunk 1 above)
Comment (inaki-amatria) [reply .github/workflows/Compile-and-run.yml]: Done! Let me know what you think :-)

Comment (daniel-otero) [reply .github/workflows/Compile-and-run.yml]: :disappointed:

Comment (daniel-otero) [reply .github/workflows/Compile-and-run.yml]: I think it was better before, sorry :sweat:

Comment (inaki-amatria) [reply .github/workflows/Compile-and-run.yml:20]: Reverted

**Pull request merged**